inst/atime/tests.R

```diff
@@ -0,0 +1,108 @@
+# A function to customize R package metadata and source files to facilitate version-specific installation and testing.
+#
+# This is specifically tailored for handling data.table which requires specific changes in non-standard files (such as the object file name in Makevars and version checking code in onLoad.R) 
+# to support testing across different versions (base and HEAD for PRs, commits associated with historical regressions, etc.) of the package. 
+# It appends a SHA1 hash to the package name (PKG.SHA), ensuring each version can be installed and tested separately.
+#
+# @param old.Package Current name of the package.
+# @param new.Package New name of the package, including a SHA hash.
+# @param sha SHA1 hash used for differentiating versions.
+# @param new.pkg.path Path to the package files.
+#
+# @details
+# The function modifies:
+# - DESCRIPTION, updating the package name.
+# - Makevars, customizing the shared object file name and adjusting the build settings.
+# - R/onLoad.R, adapting custom version checking for package loading operations.
+# - NAMESPACE, changing namespace settings for dynamic linking.
+#
+# @examples
+# pkg.edit.fun("data.table", "data.table.some_SHA1_hash", "some_SHA1_hash", "/path/to/data.table")
+#
+# @return None (performs in-place file modifications)
+# @note This setup is typically unnecessary for most packages but essential for data.table due to its unique configuration.
+pkg.edit.fun = function(old.Package, new.Package, sha, new.pkg.path) {
+      pkg_find_replace <- function(glob, FIND, REPLACE) {
+        atime::glob_find_replace(file.path(new.pkg.path, glob), FIND, REPLACE)
+      }
+      Package_regex <- gsub(".", "_?", old.Package, fixed = TRUE)
+      Package_ <- gsub(".", "_", old.Package, fixed = TRUE)
+      new.Package_ <- paste0(Package_, "_", sha)
+      pkg_find_replace(
+        "DESCRIPTION", 
+        paste0("Package:\\s+", old.Package),
+        paste("Package:", new.Package))
+      pkg_find_replace(
+        file.path("src", "Makevars.*in"),
+        Package_regex,
+        new.Package_)
+      pkg_find_replace(
+        file.path("R", "onLoad.R"),
+        Package_regex,
+        new.Package_)
+      pkg_find_replace(
+        file.path("R", "onLoad.R"),
+        sprintf('packageVersion\\("%s"\\)', old.Package),
+        sprintf('packageVersion\\("%s"\\)', new.Package))
+      pkg_find_replace(
+        file.path("src", "init.c"),
+        paste0("R_init_", Package_regex),
+        paste0("R_init_", gsub("[.]", "_", new.Package_)))
+      pkg_find_replace(
+        "NAMESPACE",
+        sprintf('useDynLib\\("?%s"?', Package_regex),
+        paste0('useDynLib(', new.Package_))
+    }
+
+# A list of performance tests.
+# 
+# Each entry in this list corresponds to a performance test and contains a sublist with three mandatory arguments:
+# - N: A numeric sequence of data sizes to vary.
```

Comment:
off-topic / thinking aloud: @tdhock have you considered extending {atime} for timing surfaces in 2 dimensions? i'm wondering if we'd benefit from `(nrow, ncol)` 2D benchmarking in some case. Now we are restricted to marginal approach (`nrow`, then `ncol`), but the interaction effects can sometimes be interesting.

Reply:
atime is inherently 1 dimensional (N varies), but you can make nrow and ncol vary with N